contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -996,16 +1004,31 @@ class tsqlCommonMutator : public TSqlParserBaseListener
 		// don't need to call does_object_name_need_delimiter() because problematic keywords are already allowed as function name
 	}
 
+	void enterExecute_body_batch(TSqlParser::Execute_body_batchContext *ctx) override
+	{
+		in_execute_body_batch = true;		
+	}
+		
+	void exitExecute_body_batch(TSqlParser::Execute_body_batchContext *ctx) override
+	{
+		in_execute_body_batch = false;	
+	}
+
 	void exitFunc_proc_name_server_database_schema(TSqlParser::Func_proc_name_server_database_schemaContext *ctx) override
 	{
 		GetCtxFunc<TSqlParser::Func_proc_name_server_database_schemaContext *> getDatabase = [](TSqlParser::Func_proc_name_server_database_schemaContext *o) { return o->database; };
 		GetCtxFunc<TSqlParser::Func_proc_name_server_database_schemaContext *> getSchema = [](TSqlParser::Func_proc_name_server_database_schemaContext *o) { return o->schema; };
 		std::string rewritten_name = rewrite_object_name_with_omitted_db_and_schema_name(ctx, getDatabase, getSchema);
 		std::string rewritten_schema_name = rewrite_information_schema_to_information_schema_tsql(ctx, getSchema);
 		if (!rewritten_name.empty())
-			rewritten_query_fragment.emplace(std::make_pair(ctx->start->getStartIndex(), std::make_pair(::getFullText(ctx), rewritten_name)));
+		{
+			int EXEC_prepend_length = (in_execute_body_batch) ? fragment_EXEC_prefix.length() : 0;  // add length of prefix prepended internally for execute_body_batch 
```

Comment:
What is a exec_preend_length ? Would you mind give some explain?

Reply:
EXEC_prepend_length is fragment_EXEC_prefix.length() (as assigned above), which is the length of the string "EXEC ". This is prepended in case of Execute_body_batch, i.e. execution of a stored proc as the first thing in a batch, so without an EXEC[UTE] keyword. We need to prepend EXEC but need to compensate for the additional length thus inserted or the rewriting will not work.